source/slang/slang-ir-operator-shift-overflow.cpp

```diff
@@ -0,0 +1,66 @@
+// slang-ir-operator-shift-overflow.cpp
+#include "slang-ir-operator-shift-overflow.h"
+
+#include "../../slang.h"
+#include "slang-ir.h"
+#include "slang-ir-insts.h"
+#include "slang-ir-layout.h"
+
+namespace Slang {
+
+    class DiagnosticSink;
+    struct IRModule;
+
+    void checkForOperatorShiftOverflowRecursive(
+        IRInst* inst,
+        CompilerOptionSet& optionSet,
+        DiagnosticSink* sink)
+    {
+        if (auto code = as<IRGlobalValueWithCode>(inst))
```

Comment:
in some code,  i noticed that we have patterns like 
auto code = as<IRGlobalValueWithCode>(inst);
if (!code)
 continue;

this is done to keep the nesting managable. but I leave this up to you.

Reply:
Aha, you are talking about the indentation.
You are right, I can change it here.

FYI, this line was copied from another file, slang-ir-missing-return.cpp.

Reply:
It turned out I cannot do it for this case.
There is another block down there that needs to be executed regardless code is null pointer or not.